arch/x86/zefi/zefi.py

```diff
@@ -107,8 +107,8 @@ def build_elf(elf_file):
     #  + UEFI can take interrupts on our stack, so no red zone
     #  + UEFI API assumes 16-bit wchar_t
     cmd = [args.compiler, "-shared", "-Wall", "-Werror", "-I.",
-        "-fno-stack-protector", "-fpic", "-mno-red-zone", "-fshort-wchar",
-        "-Wl,-nostdlib", "-T", ldscript, "-o", "zefi.elf", cfile]
+           "-I", include_dir, "-fno-stack-protector", "-fpic", "-mno-red-zone",
+           "-fshort-wchar", "-Wl,-nostdlib", "-T", ldscript, "-o", "zefi.elf", cfile]
```

Comment:
Nothing wrong with this update, but a small proposal worth consider.

CMake is our build system, and is designed for handling compilation and linking.

So maybe it should be considered to change this script into generating source code, and then use that source code for an extra build target which generates the `zefi.elf`.

Some pros on such change:
- CMake can make sure `GENERATED` property is correctly set on the generated source.
- No need to also pass include folder(s) to the script.
- it is clear what happens during build, as the compilation and linking of `zefi.c` and `zefi.elf` will be part of `build.ninja` / `Makefile`. Thus also show up with `ninja -v`.
- The source file will be part of `compile_commands.json`, used by IDEs.
- zefi.elf will be dedicated target, meaning it will be possible to execute `ninja zefi.elf` / `make zefi,ekf` directly, and thus the target is built including its dependencies.

Reply:
Looks like an enhancement to me. Definitely worth doing but not in this PR